activejob/lib/active_job/test_helper.rb

```diff
@@ -694,6 +694,10 @@ def flush_enqueued_jobs(only: nil, except: nil, queue: nil, at: nil)
 
       def prepare_args_for_assertion(args)
         args.dup.tap do |arguments|
+          if arguments[:queue].is_a?(Symbol)
+            arguments[:queue] = arguments[:queue].to_s
+          end
```

Comment:
```suggestion
          arguments[:queue] = arguments[:queue].to_s
```

This is a great find ✨ 

I don't think this needs a type check, every value should be coerced to a string. This is because, by my reading, `ActiveJob::Base` will always coerce the queue name to a string via `Array#join`:

https://github.com/rails/rails/blob/70d3b8760fb625d0ebc8b2d0482c45645d3bb7b8/activejob/lib/active_job/queue_name.rb#L48-L52

Blind coercion to a string is also how a different part of the test helper compares queue name too:

https://github.com/rails/rails/blob/744b6715a5d07a2ddffceea8283f7b787541bc60/activejob/lib/active_job/test_helper.rb#L660-L660

Reply:
Hey! @bensheldon Thanks a lot for looking at my PR.

I did try converting the queue name to a string directly like you suggested `arguments[:queue] = arguments[:queue].to_s` but the tests start failing.

Here's why:
1. If we don't pass a queue name, `arguments[:queue]` will be `nil` and converting that to string will give us an empty string `""` , causing the test to fail because `"" != "default"`.
2. Apparently, `arguments[:queue]` can also be a `Proc`. The below test and another similar test (at line 1894) fail because of this.

https://github.com/rails/rails/blob/744b6715a5d07a2ddffceea8283f7b787541bc60/activejob/test/cases/test_helper_test.rb#L576-L596

We can fix the 1st problem I mentioned by doing this:
```rb
arguments[:queue] = arguments[:queue].to_s if arguments.key?(:queue)
```

To fix the 2nd one, We have to rely on type-checking, hence the current change.

Let me know if there's better way :)

Reply:
oh! I see what you mean 🤔

What about?

```ruby
if !arguments[:queue].nil? && !arguments[:queue].respond_to?(:call)
  arguments[:queue] = arguments[:queue].to_s 
end

```

I don't really like it aesthetically though....

I think then I'm ok with specifically checking for Symbol because I imagine that will cover the sharp edge most people will experience; I think the case I'm imagining is if someone tries to match a numeric (e.g. `queue: 99` instead of `queue: "99"`) but that seems really rare. 

👍🏻 to your change (I'm not a Rails approver, I just keep a real close eye on Active Job).